examples/step-1/doc/intro.dox

```diff
@@ -137,22 +137,47 @@ later on. Additionally, as of deal.II version 9.0, GridGenerator::hyper_shell()
-some elements of simple loops over all cells. We create two grids, one
-which is a regularly refined square (not very exciting, but a common
-starting grid for some problems), and one more geometric attempt: a
-ring-shaped domain, which is refined towards the inner edge. Through
-this, you will get to know three things every finite element program
-will have to have somewhere: An object of type Triangulation for the
-mesh; a call to the GridGenerator functions to generate a mesh; and
-loops over all cells that involve iterators (iterators are a
-generalization of pointers and are frequently used in the C++ standard
-library; in the context of deal.II, the @ref Iterators module talks
-about them).
+Let's come back to step-1, the current program. The goal of this
+program is to introduce you to the Triangulation class that is at the
+core of every finite element program. The name "triangulation" in this
+context is mostly historical: To finite element practitioner, the
+terms "triangulation", "mesh", and "grid" are all synonymous and
+describe a subdivision of a domain on which a differential equation is
+posed into cells of some kind. If the domain is two-dimensional, these
+cells may indeed be triangles, but they could also be quadrilaterals
+(four-sided objects such as squares and rectangles, and their
+deformations). In one space dimension, the cells are line segments. In
+three space dimensions, they can be tetrahedra, hexahedra (deformed
+cubes), pyramids (a four-sided base with three triangles connecting to
+a point at the top), and "wedges" (two triangles at the bottom and
+top, connected by three quadrilaterals). Collections of any such cells are
+"triangulations" in common usage of the word even though they may not
+actually have triangles in them. All of them are also "grids" in
+common usage of the word even though the vertices are in neat rows
+parallel to the coordinate axes. In other words, whenever you read any
+of the three terms in the tutorials or the library's documentation,
+consider them equivalent.
+
+What this program shows, then, is how to create a Triangulation
+object, and to operate on it. The underlying concept of a
+Triangulation is that it is a *container*, i.e., a class that stores a
+*collection of cells*. As is common in modern programming languages,
+the key operation on containers is that one can *iterate* over its
+elements, and that's exactly what we will do below.
+
+Specifically, in the program we create two grids, one which is a
+regularly refined square (not very exciting, but a common starting
+grid for many problems), and one that is a more geometric attempt: a
+ring-shaped domain that is refined towards the inner edge. The process
+of refining the mesh in this way will illustrate how we iterate (i.e.,
+loop) over the elements of the triangulation (i.e., the cells of the
+mesh). You will see many many more such loops throughout the remainder
```

Comment:
I would drop the second "many", but I realize it is personal taste, so feel free to decide what you like best:
```suggestion
mesh). You will see many more such loops throughout the remainder
```